internals/overlord/checkstate/manager_test.go

```diff
@@ -869,3 +868,54 @@ func (s *ManagerSuite) TestReplan(c *C) {
 	c.Assert(status, Matches, "Do.*")
 	c.Assert(change.Kind(), Equals, "perform-check")
 }
+
+func (s *ManagerSuite) TestMetrics(c *C) {
+	origLayer := &plan.Layer{
+		Checks: map[string]*plan.Check{
+			"chk1": {
+				Name:      "chk1",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: 10 * time.Millisecond},
+				Timeout:   plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk1"},
+				Startup:   plan.CheckStartupDisabled,
+			},
+		},
+	}
+	err := s.planMgr.AppendLayer(origLayer, false)
+	c.Assert(err, IsNil)
+
+	// Ensure and wait for the counter to increase.
+	s.manager.PlanChanged(&plan.Plan{
+		Checks: map[string]*plan.Check{
+			"chk1": {
+				Name:      "chk1",
+				Override:  "replace",
+				Period:    plan.OptionalDuration{Value: 10 * time.Millisecond},
+				Timeout:   plan.OptionalDuration{Value: time.Second},
+				Threshold: 3,
+				Exec:      &plan.ExecCheck{Command: "echo chk1"},
+				Startup:   plan.CheckStartupEnabled,
+			},
+		},
+	})
+	waitChecks(c, s.manager, []*checkstate.CheckInfo{
+		{Name: "chk1", Startup: "enabled", Status: "up", Threshold: 3, PerformCheckCount: 2, RecoverCheckCount: 0},
```

Comment:
Hmm, yes, we definitely shouldn't be getting these fields coming through here. In fact, we should either have an internal `checkInfo` (with the metrics fields) and an exported one `CheckInfo` without them, or we should explicitly set them to zero after we've made the copy in `Checks()` -- probably the latter is simplest.

For testing, we should call the `WriteMetrics` method with a fake `MetricsWriter` that just appends to a slice or something, and compare that. Or just use an OpenTelemetryWriter and compare the output -- I see you're already doing that for the services metrics.